net/local/local_release.c

```diff
@@ -72,18 +72,17 @@ int local_release(FAR struct local_conn_s *conn)
   if (conn->lc_state == LOCAL_STATE_LISTENING)
     {
       FAR struct local_conn_s *accept;
-      FAR dq_entry_t *waiter;
+      FAR dq_entry_t *waiter = dq_peek(&conn->u.server.lc_waiters);
 
       DEBUGASSERT(conn->lc_proto == SOCK_STREAM);
 
       /* Are there still clients waiting for a connection to the server? */
 
-      for (waiter = dq_peek(&conn->u.server.lc_waiters);
-           waiter != NULL;
-           waiter = dq_next(&accept->u.accept.lc_waiter))
+      while (waiter != NULL)
```

Comment:
please upstream `dq_for_every_safe()` and replace to safe version

https://github.com/search?q=repo%3Aapache%2Fnuttx%20dq_for_every_safe&type=code

Reply:
@anchao  Done.